Library/ViewModels/NoShippingPledgeViewModel.swift

```diff
@@ -987,23 +964,51 @@ public class NoShippingPledgeViewModel: NoShippingPledgeViewModelType, NoShippin
     self.pledgeOverTimeConfigData = Signal.combineLatest(
       self.showPledgeOverTimeUI,
       project,
-      pledgeTotal
+      pledgeTotal,
+      self.paymentPlanSelectedSignal
     )
-    .filter { showPledgeOverTimeUI, _, _ in
-      showPledgeOverTimeUI
-    }
-    .map { _, project, pledgeTotal -> PledgePaymentPlansAndSelectionData in
+    .map { showUI, project, pledgeTotal, planSelected -> PledgePaymentPlansAndSelectionData? in
+      guard showUI else { return nil }
       // TODO: temporary code to simulate the ineligible state. Implementation [MBL-1838](https://kickstarter.atlassian.net/browse/MBL-1838)
       let isIneligible = pledgeTotal < 150
 
       return PledgePaymentPlansAndSelectionData(
-        selectedPlan: .pledgeInFull,
+        selectedPlan: planSelected,
         increments: mockPledgePaymentIncrement(),
         ineligible: isIneligible,
         project: project
       )
     }
 
+    self.configurePledgeRewardsSummaryViewWithData = Signal.combineLatest(
+      initialData,
+      pledgeTotal,
+      additionalPledgeAmount,
+      shippingSummaryViewData,
+      rewards,
+      self.pledgeOverTimeConfigData
```

Comment:
it looks like this signal will only emit if the PLOT ui is visible. (If one signal from `combineLatest` never emits, the input will never all exist and the signal won't do anything.) You'll want to make sure this emits `nil` when the UI doesn't show.

Reply:
Confirmed! I ensured that the signal emits even when the input is nil, and I verified it works as expected. Let me know if there’s anything else you’d like me to double-check.